### PR TITLE
Add ability to push vars and events in form submits

### PR DESCRIPTION
**Pull request by briansage**
This works in our Picreel staging environment. This PR is mostly a double-check for how I'm refactoring the test to make this workable in QUnit.

Quickly falling out of love with QUnit for testing growing libraries. Wishing I had started with Mocha.

### CODE REVIEW

File: lib/factories/subscriber.js [hunk 1]
```diff
@@ -83,6 +85,10 @@
       var data = {};
 
       data['email'] = this.$(this.emailSelector).val();
+      if(this.vars.length)
```
Comment (rickyhanna) [inline lib/factories/subscriber.js]: I'd prefer these one-line if's to be one line [ double :) ]

File: lib/factories/subscriber.js [hunk 2]
```diff
@@ -19,6 +19,8 @@
       this.onerror        = this.onerror         || function() {};
```
Comment (rickyhanna) [inline lib/factories/subscriber.js]:
```
var noop = function() {};
```

Then you can reuse it :)

Comment (briansage) [inline lib/factories/subscriber.js]: I like it. Change pushed.

**Comment by rickyhanna** [comment]
:cake:

**Pull request merged**